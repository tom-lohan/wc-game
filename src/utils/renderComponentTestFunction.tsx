import { configureStore } from '@reduxjs/toolkit'
import { render } from '@testing-library/react'
import React, { ReactNode } from 'react'
import { Provider } from 'react-redux'
import '@testing-library/jest-dom'
import { WCGInitialState } from '../wcgSlice'
import gameReducer from '../wcgSlice'

export const renderComponentTestFunction = (
    children: ReactNode,
    modifiedStore: object = {}
) => {
    const store = {
        wcgame: { ...WCGInitialState },
    }

    const mergedStore = { ...store, ...modifiedStore }

    const mockedStore = configureStore({
        reducer: {
            wcgame: gameReducer,
        },
        preloadedState: mergedStore,
        middleware: (getDefaultMiddleware) =>
            getDefaultMiddleware({ serializableCheck: false }),
    })

    return render(<Provider store={mockedStore}>{children}</Provider>)
}
