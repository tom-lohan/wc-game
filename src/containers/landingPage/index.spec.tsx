import React from 'react'
import '@testing-library/jest-dom'
import { renderComponentTestFunction } from '../../utils/renderComponentTestFunction'
import { LandingPage } from '.'

const mockedSetSelectedPage = jest.fn()

describe('LandingPage', () => {
    beforeEach(() => {
        jest.clearAllMocks()
        jest.restoreAllMocks()
    })

    it('should render correctly', () => {
        const { container } = renderComponentTestFunction(
            <LandingPage setSelectedPage={mockedSetSelectedPage} />
        )
        expect(container).toMatchSnapshot()
    })
})
