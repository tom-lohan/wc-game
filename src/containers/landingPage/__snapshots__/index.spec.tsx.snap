// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should render correctly 1`] = `
<div>
  <main
    class="landingPage flex flex-col justify-center items-center"
    data-testid="landing_page"
  >
    <h1
      class="title px-5"
    >
      World Cup 2022 Sweepstakes
    </h1>
    <button
      class="ant-btn ant-btn-default ant-btn-lg"
      type="button"
    >
      <span>
        Get started
      </span>
    </button>
  </main>
</div>
`;
